modules/tinymce/src/themes/silver/test/ts/browser/editor/core/UndoRedoUiTest.ts

```diff
@@ -0,0 +1,59 @@
+import { UiFinder, Waiter } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { SugarBody } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.themes.silver.editor.core.UndoRedoUiTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    toolbar: 'redo undo',
+    menubar: true
+  }, []);
+
+  const assertToolbarButtonDisabled = (title: string, state: boolean = true) =>
+    UiFinder.exists(SugarBody.body(), `button[title="${title}"][aria-disabled="${state}"]`);
+
+  const pAssertMenuItemDisabled = (item: string, state: boolean = true) =>
+    Waiter.pTryUntil('Wait for a specific menu to open', () => UiFinder.exists(SugarBody.body(), `[role="menuitem"][aria-disabled=${state}"]:contains("${item}")`));
+
+  it('TINY-8101: Undo/redo should function as expected in normal editor mode', () => {
+    const editor = hook.editor();
+    assertToolbarButtonDisabled('Redo');
+    assertToolbarButtonDisabled('Undo');
+
+    TinyUiActions.clickOnMenu(editor, 'button:contains("Edit")');
+    pAssertMenuItemDisabled('Redo');
+    pAssertMenuItemDisabled('Undo');
+
+    // 1. Insert content, expect undo to be enabled
+    editor.insertContent('<p>slsl</p>');
+    assertToolbarButtonDisabled('Redo');
+    assertToolbarButtonDisabled('Undo', false);
+
+    TinyUiActions.clickOnMenu(editor, 'button:contains("Edit")');
```

Comment:
This likely isn't going to do what you're expecting because you never close the menu... so you're actually trying to open it twice here. So we probably should be re-focusing the editor after doing the assertions to ensure the menu closes.